### PR TITLE
List Port Forwardings

### PR DESCRIPTION
**Pull request by simonre**
Prior to a PR being reviewed, there needs to be a Github issue that the PR
addresses. Replace the brackets and text below with that issue number.

For #1637 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

Source: 
https://github.com/openstack/neutron/blob/f21444f247108212dc835036c98e354d6e6f7ace/neutron/services/portforwarding/pf_plugin.py#L487

API:
https://docs.openstack.org/api-ref/network/v2/?expanded=create-port-forwarding-detail#list-floating-ip-port-forwardings

### CODE REVIEW

**Comment by simonre** [comment]
@jtopjian This is ready for review

File: openstack/networking/v2/extensions/layer3/portforwarding/requests.go [hunk 1]
```diff
@@ -1,6 +1,53 @@
...
+type ListOptsBuilder interface {
+	ToPortForwardingListQuery() (string, error)
+}
+
+// ListOpts allows the filtering and sorting of paginated collections through
+// the API. Filtering is achieved by passing in struct field values that map to
+// the port forwarding attributes you want to see returned. SortKey allows you to
+// sort by a particular network attribute. SortDir sets the direction, and is
+// either `asc' or `desc'. Marker and Limit are used for pagination.
+type ListOpts struct {
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/layer3/portforwarding/requests.go]: We need to add `limit` (int) and `marker` (string) here.

Gophercloud's paging automatically sets limit and marker when paging internally, but if the user wanted to limit the results themselves, we need that in `ListOpts`.

**Comment by simonre** [comment]
@jtopjian I added the `limit` and `marker` fields

**Comment by jtopjian** [comment]
@simonre sorry for the delay - I'm out of the office. Thank you!

**Pull request merged**